flutter/lib/state/task_runner.dart

```diff
@@ -0,0 +1,348 @@
+
+  final Benchmark benchmark;
+  final BenchmarkRunMode runMode;
+  final String logDir;
+  late final RunSettings runSettings;
+
+  _NativeRunHelper({
+    required this.enableArtificialLoad,
+    required this.backendBridge,
+    required this.benchmark,
+    required this.runMode,
+    required bool isTestMode,
+    required ResourceManager resourceManager,
+    required List<pb.Setting> commonSettings,
+    required String backendLibName,
+    required String logParentDir,
+  }) : logDir = '$logParentDir/${benchmark.id}-${runMode.readable}' {
+    runSettings = benchmark.createRunSettings(
+      runMode: runMode,
+      resourceManager: resourceManager,
+      commonSettings: commonSettings,
+      backendLibName: backendLibName,
+      logDir: logDir,
+      isTestMode: isTestMode,
+    );
+  }
+
+  Future<RunInfo> run() async {
+    await Directory(logDir).create(recursive: true);
+
+    Cancelable<void>? artificialLoadHandle;
+    if (enableArtificialLoad) {
+      print('Apply the artificial CPU load for ${benchmark.taskConfig.id}');
+      const value = 999999999999999.0;
+      artificialLoadHandle =
+          Executor().execute(arg1: value, fun1: _doSomethingCPUIntensive);
+    }
+
+    try {
+      return await _invokeNativeRun();
+    } finally {
+      if (artificialLoadHandle != null) {
+        artificialLoadHandle.cancel();
+      }
+    }
+  }
+
+  Future<RunInfo> _invokeNativeRun() async {
+    final logPrefix = '${benchmark.id}: $runMode mode';
+
+    print('$logPrefix: starting...');
+    final stopwatch = Stopwatch()..start();
+    final nativeResult = await backendBridge.run(runSettings);
+    final elapsed = stopwatch.elapsed;
+    print('$logPrefix: elapsed: $elapsed');
+    print('$logPrefix: result: $nativeResult');
+
+    if (!_checkAccuracy(nativeResult)) {
+      throw '$logPrefix: accuracy is invalid (backend may be corrupted)';
+    }
+
+    final runInfo = await _makeRunInfo(nativeResult);
+    print('$logPrefix: throughput: ${runInfo.throughput}');
+
+    return runInfo;
+  }
+
+  Future<RunInfo> _makeRunInfo(NativeRunResult nativeResult) async {
+    final loadgenInfo = await extractLoadgenInfo();
+    final throughput = _calculateThroughput(nativeResult, loadgenInfo);
+
+    return RunInfo(
+      settings: runSettings,
+      result: nativeResult,
+      loadgenInfo: loadgenInfo,
+      throughput: throughput,
+    );
+  }
+
+  bool _checkAccuracy(NativeRunResult result) {
+    return (result.accuracy1?.isInBounds() ?? true) &&
+        (result.accuracy2?.isInBounds() ?? true);
+  }
+
+  double? _calculateThroughput(
+    NativeRunResult result,
+    LoadgenInfo? loadgenInfo,
+  ) {
+    if (loadgenInfo == null) {
+      return null;
+    }
+    if (benchmark.info.isOffline) {
+      return result.numSamples / loadgenInfo.latency90;
+    } else {
+      return 1.0 / loadgenInfo.latency90;
+    }
+  }
+
+  Future<LoadgenInfo?> extractLoadgenInfo() async {
+    const logFileName = 'mlperf_log_detail.txt';
```

Comment:
I think we can make all string `const` like this `static const`.

Reply:
`const` keyword in Dart is not like `const` in C++. It's already `static const`, because in Dart all const values use static initialization, and can only have one instance.
`static const varName` in code will result in compilation error, because it's not a valid syntax in Dart for local fields (for class fields it's the other way around).

Reply:
Yeah, `static const` mean the same as `class const` in some other languages, so it needs to be defined as a class field. Normally, I would either define this kind of string as a global constant, or a class constant, since it is kind of setting.

Reply:
OK, I'll move it to class level